tests/unit/s2n_client_supported_versions_extension_test.c

```diff
@@ -396,6 +392,62 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
 
+    /**
+     *= https://tools.ietf.org/rfc/rfc8446#appendix-D.2
+     *= type=test
+     *# A TLS server can also receive a ClientHello indicating a version number smaller than its highest supported
+     *# version. If the "supported_versions" extension is present, the server MUST negotiate using that extension as
+     *#  described in Section 4.2.1.
+     */
+    if (s2n_is_tls13_fully_supported()) {
+        {
+            DEFER_CLEANUP(struct s2n_cert_chain_and_key * chain_and_key,
+                    s2n_cert_chain_and_key_ptr_free);
+            EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                    S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN,
+                    S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
+
+            DEFER_CLEANUP(struct s2n_config *configNew = s2n_config_new(),
+                    s2n_config_ptr_free);
+            EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(configNew, chain_and_key));
+
+            DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(server_conn);
+            EXPECT_SUCCESS(s2n_connection_set_config(server_conn, configNew));
+
+            DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            EXPECT_NOT_NULL(client_conn);
+            EXPECT_SUCCESS(s2n_connection_set_config(client_conn, configNew));
+
+            struct s2n_test_io_pair io_pair = { 0 };
+            EXPECT_SUCCESS(s2n_io_pair_init_non_blocking(&io_pair));
+            EXPECT_SUCCESS(s2n_connections_set_io_pair(client_conn, server_conn, &io_pair));
+
+            struct s2n_stuffer *hello_stuffer = NULL;
+            hello_stuffer = &client_conn->handshake.io;
+            EXPECT_SUCCESS(s2n_client_hello_send(client_conn));
+
+            /* Overwrite the Client Hello protocol version to TLS10 */
+            uint8_t small_protocol_version[S2N_TLS_PROTOCOL_VERSION_LEN] = {0};
+            small_protocol_version[0] = S2N_TLS10 / 10;
+            small_protocol_version[1] = S2N_TLS10 % 10;
+
+            EXPECT_SUCCESS(s2n_stuffer_rewrite(hello_stuffer));
+            EXPECT_SUCCESS(s2n_stuffer_write_bytes(hello_stuffer, small_protocol_version, S2N_TLS_PROTOCOL_VERSION_LEN));
+            EXPECT_SUCCESS(s2n_stuffer_write(&server_conn->handshake.io, &hello_stuffer->blob));
+
+            /* The server does not use the protocol version in the Client Hello to set the actual protocol version. */
```

Comment:
Can you move this comment to line 443? It's explaining what those assertions mean.